This patch enables IPv4 Link-Local address fallback of dhcpcd, so an interface with enabled
ipv4.method=auto will fallback to link-local addressing if not DHCP server was reachable
diff -aurN a/src/dhcp/nm-dhcp-dhcpcd.c b/src/dhcp/nm-dhcp-dhcpcd.c
--- a/src/dhcp/nm-dhcp-dhcpcd.c	2020-05-29 16:21:10.000000000 +0200
+++ b/src/dhcp/nm-dhcp-dhcpcd.c	2021-01-15 10:52:55.894771226 +0100
@@ -105,11 +105,6 @@
 
 	g_ptr_array_add (argv, (gpointer) "-K");    /* Disable built-in carrier detection */
 
-	g_ptr_array_add (argv, (gpointer) "-L");    /* Disable built-in IPv4LL */
-
-	/* --noarp. Don't request or claim the address by ARP; this also disables IPv4LL. */
-	g_ptr_array_add (argv, (gpointer) "-A");
-
 	g_ptr_array_add (argv, (gpointer) "-G");    /* Let NM handle routing */
 
 	g_ptr_array_add (argv, (gpointer) "-c");    /* Set script file */
